Userland/Libraries/LibJS/Runtime/AbstractOperations.cpp

```diff
@@ -1223,7 +1223,7 @@ CanonicalIndex canonical_numeric_index_string(PropertyKey const& property_key, C
 
     // FIXME: We return 0 instead of n but it might not observable?
     // 3. If SameValue(! ToString(n), argument) is true, return n.
-    if (double_value.to_deprecated_string_without_side_effects() == argument)
+    if (TRY_OR_THROW_OOM(vm, double_value.to_string_without_side_effects()).bytes_as_string_view() == argument)
```

Comment:
Just call `number_to_string()` directly here instead of going `double -> JS::Value -> String`